pkgs/development/libraries/pmix/default.nix

```diff
@@ -46,6 +52,26 @@ stdenv.mkDerivation rec {
 
   postInstall = ''
     find $out/lib/ -name "*.la" -exec rm -f \{} \;
+
+    moveToOutput "bin/pmix_info" "''${!outputDev}"
+    moveToOutput "bin/pmixcc" "''${!outputDev}"
+    moveToOutput "share/pmix/pmixcc-wrapper-data.txt" "''${!outputDev}"
+
+    # The path to the pmixcc-wrapper-data.txt is hard coded and
+    # points to $out instead of dev. Use wrapper to fix paths.
+    wrapProgram $dev/bin/pmixcc \
+      --set PMIX_INCLUDEDIR $dev/include \
+      --set PMIX_PKGDATADIR $dev/share/pmix
+  '';
+
+  postFixup = ''
+    # The build info (parameters to ./configure) are hardcoded
+    # into the library. This clears all references to $dev/include.
+    remove-references-to -t $dev $(readlink -f $out/lib/libpmix.so)
+
+    # Fix compiler to the one pmix was built with
```

Comment:
The comment is a bit misleading, it's not the one used for the actual "build"

Reply:
What do you suggest instead?

Reply:
Well you're just pinning at build time the `host->target` compiler to be used at runtime. The "why" is clear: because otherwise they just expect the compiler magically to be "there", so you have to put some value there. "Which" value should it be? Maybe add a comment explaining the offsets/why you chose `targetPackages`. Could be a reference to the manual. Also not obvious what the old values were: did they use FHS paths or `PATH`-relative names?

Reply:
This is the same logic as in openmpi, which was introduced here: https://github.com/NixOS/nixpkgs/pull/58964#discussion_r275059427.
The original just says "gcc", which would work in most cases. However, fixing the compiler at build time guarantees that your whatever build with the wrapper is build with the same compiler version as pmix (unless you chose to deliberately override the compiler).
The fact that you use `targetPackages` instead of just `stdenv.cc` is supposed to make it cross compiler proof. Although I have never tried to cross build any of the packages of the MPI ecosystem.

Reply:
> The fact that you use targetPackages instead of just stdenv.cc is supposed to make it cross compiler proof

I think this comment would be more helpful checked in git, than the current comment is. E.g. just briefly mention that `stdenv.cc` is `build->host`, but at runtime we need `host->target`, so we take `targetPackages.stdenv.cc`